driver-mct/main/cime_comp_mod.F90

```diff
@@ -4708,6 +4705,17 @@ subroutine cime_run_calc_budgets2(in_cplrun)
              call seq_diagBGC_ice_mct(ice(ens1), fractions_ix(ens1), infodata, do_i2x=.true.)
           endif
        endif
+       if (lnd_present) then
```

Comment:
You could put these if blocks together with if(do_bgc_budgets) as the outer if.  Up to you.

Reply:
Thanks for the suggestion @rljacob. It is cleaner that way